### ISSUE TITLE
[WatchESPN] Update URL Support

### ISSUE THREAD

**Issue by IONECarter** [yt-dlp/yt-dlp#4362]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.06.29** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

US

### Provide a description that is worded well enough to be understood

ESPN uses a number of different URL structures for their content. I've noticed that some ESPN+ content uses the structure `https://www.espn.com/espnplus/player/_/id/xxxx`. This is currently not picked up be an extractor an results in an error (**Log 1** below)however, if you change `espnplus` to `watch` it gets picked up by the `WatchESPNIE` extractor and works fine (**Log 2** below).

I though it could be as simple as changing the respective part of the `WatchESPNIE` regex to be `(watch|espnplus)`, but that didn't appear to work. Assuming someone more familiar with the code may know how to account for this.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
**Log 1**
[debug] Command-line config: ['-vU', 'https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\\#bucketId\\=41432', '--cookies-from-browser', 'firefox', '--ap-mso', 'Comcast_SSO', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', '-F']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.06.29 [9d339c4]
[debug] Python 3.9.13 (CPython 64bit) - macOS-12.4-x86_64-i386-64bit
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "XXXX.Default User/cookies.sqlite"
[Cookies] Extracted X cookies from firefox
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/download/2022.06.29/_update_spec
Latest version: 2022.06.29, Current version: 2022.06.29
yt-dlp is up to date (2022.06.29)
[debug] [ESPNArticle] Extracting URL: https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432
[ESPNArticle] 317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432: Downloading webpage
ERROR: [ESPNArticle] 317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432: Unable to extract video id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 640, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/espn.py", line 207, in _real_extract
    video_id = self._search_regex(
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 1192, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)```


**Log 2**
```[debug] Command-line config: ['-vU', 'https://www.espn.com/watch/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\\#bucketId\\=41432', '--cookies-from-browser', 'firefox', '--ap-mso', 'Comcast_SSO', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', '-F']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.06.29 [9d339c4]
[debug] Python 3.9.13 (CPython 64bit) - macOS-12.4-x86_64-i386-64bit
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "XXXX.Default User/cookies.sqlite"
[Cookies] Extracted X cookies from firefox
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/download/2022.06.29/_update_spec
Latest version: 2022.06.29, Current version: 2022.06.29
yt-dlp is up to date (2022.06.29)
[debug] [WatchESPN] Extracting URL: https://www.espn.com/watch/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 317f5fd1-c78a-4ebe-824a-129e0d348421:
ID       EXT RESOLUTION FPS │   TBR PROTO │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────
hls-489  mp4 384x216     29 │  489k m3u8  │ avc1.4d401f  489k mp4a.40.2  0k
hls-847  mp4 512x288     29 │  847k m3u8  │ avc1.4d401f  847k mp4a.40.2  0k
hls-1251 mp4 640x360     29 │ 1251k m3u8  │ avc1.4d401f 1251k mp4a.40.2  0k
hls-1864 mp4 896x504     29 │ 1864k m3u8  │ avc1.4d401f 1864k mp4a.40.2  0k
hls-2604 mp4 960x540     59 │ 2604k m3u8  │ avc1.4d4020 2604k mp4a.40.2  0k
hls-3621 mp4 1280x720    59 │ 3621k m3u8  │ avc1.640028 3621k mp4a.40.2  0k
hls-5763 mp4 1280x720    59 │ 5763k m3u8  │ avc1.640028 5763k mp4a.40.2  0k```
```

**Comment by dirkf** [comment]
Do you have a log of the modified extractor failing?

Otherwise, this patch, which is basically yours, ought to work, given what you report:
```diff 
--- old/yt_dlp/extractor/espn.py
+++ new/yt_dlp/extractor/espn.py
@@ -281,7 +281,7 @@
 
 
 class WatchESPNIE(AdobePassIE):
-    _VALID_URL = r'https://www.espn.com/watch/player/_/id/(?P<id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})'
+    _VALID_URL = r'https?://(?:www\.)?espn\.com/(?:watch|espnplus)/player/_/id/(?P<id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})'
     _TESTS = [{
         'url': 'https://www.espn.com/watch/player/_/id/ba7d17da-453b-4697-bf92-76a99f61642b',
         'info_dict': {
```

**Comment by ischmidt20** [comment]
As the author of the original extractor, I figured I would be leaving out matches with the regex. Someone should submit a pull request with @dirkf's fix and maybe an additional test and I presume it'll get approved

**Comment by pukkandan** [comment]
If anyone can confirm this change is sufficient, I can directly merge it

**Comment by IONECarter** [comment]
Thanks all for the quick feedback. Unfortunately, using the regex in @dirkf patch doesn't seem to work and provides the following.
```
[debug] Command-line config: ['-vU', 'https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\\#bucketId\\=41432', '--cookies-from-browser', 'firefox', '--ap-mso', 'Comcast_SSO', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', '-F']
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/download/2022.06.29/_update_spec
Latest version: 2022.06.29, Current version: 2022.06.29
yt-dlp is up to date (2022.06.29)
[debug] [ESPNArticle] Extracting URL: https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432
[ESPNArticle] 317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432: Downloading webpage
ERROR: [ESPNArticle] 317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432: Unable to extract video id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 640, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/espn.py", line 207, in _real_extract
    video_id = self._search_regex(
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 1192, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
```
That is consistent with the error I got when modifying it initially to `_VALID_URL = r'https://www.espn.com/(watch|espnplus)/player/_/id/(?P<id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})'` which provides
```
[debug] Command-line config: ['-vU', 'https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\\#bucketId\\=41432', '--cookies-from-browser', 'firefox', '--ap-mso', 'Comcast_SSO', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', '-F']
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/download/2022.06.29/_update_spec
Latest version: 2022.06.29, Current version: 2022.06.29
yt-dlp is up to date (2022.06.29)
[debug] [ESPNArticle] Extracting URL: https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432
[ESPNArticle] 317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432: Downloading webpage
ERROR: [ESPNArticle] 317f5fd1-c78a-4ebe-824a-129e0d348421\#bucketId\=41432: Unable to extract video id; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 640, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/espn.py", line 207, in _real_extract
    video_id = self._search_regex(
  File "/usr/local/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 1192, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
```

**Comment by Lesmiscore** [comment]
Try removing after `#`

**Comment by dirkf** [comment]
That's completely weird. In either case of the replacement `_VALID_URL` tested, `WatchESPNIE.suitable('https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\\#bucketId\\=41432')` should be `True`, as in 
```
>>> re.match(r'https?://(?:www\.)?espn\.com/(?:watch|espnplus)/player/_/id/(?P<id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})', 'https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421\\#bucketId\\=41432').groupdict()
{'id': '317f5fd1-c78a-4ebe-824a-129e0d348421'}
```
and so `ESPNArticleIE.suitable(...)` must be `False` based on l.198ff of the module:
```py
    @classmethod
    def suitable(cls, url):
        return False if (ESPNIE.suitable(url) or WatchESPNIE.suitable(url)) else super().suitable(url)
```
It seems that neither patch could have been active when the code ran: both tracebacks match the original one and show `ESPNArticleIE` being used.

Does macOS Monterey "protect" files in `/usr/local/lib` in some unexpected way like Windows [System] File Protection?

**Comment by pukkandan** [comment]
@IONECarter You have lazy extractors enabled. You need to either rebuild or delete lazy extractors after modifying the files.

PS: I believe you are modifying the PIP install directly. This is not recommended to do (not just for yt-dlp, but in general). The "right" way to make manual changes is to download/clone the source code

**Comment by IONECarter** [comment]
Again thank you all, appreciate you bearing with me as I'm still learning and trying to get familiar with all this. I can confirm that once I removed `lazy_extractors.py` the patch from @dirkf worked perfectly (log below). Also, if needed, below is an additional test covering the new URL.

```
[debug] Command-line config: ['-vU', 'https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421', '--cookies-from-browser', 'firefox', '--ap-mso', 'Comcast_SSO', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', '-F']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.06.29 [9d339c4]
[debug] Lazy loading extractors is disabled
[debug] Python 3.10.5 (CPython 64bit) - macOS-12.4-x86_64-i386-64bit
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 5.0.1 (setts), ffprobe 5.0.1
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[Cookies] Extracting cookies from firefox
[debug] Extracting cookies from: "XXX.Default User/cookies.sqlite"
[Cookies] Extracted X cookies from firefox
[debug] Proxy map: {}
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
[debug] Downloading _update_spec from https://github.com/yt-dlp/yt-dlp/releases/download/2022.06.29/_update_spec
Latest version: 2022.06.29, Current version: 2022.06.29
yt-dlp is up to date (2022.06.29)
[debug] [WatchESPN] Extracting URL: https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading JSON metadata
[WatchESPN] 317f5fd1-c78a-4ebe-824a-129e0d348421: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 317f5fd1-c78a-4ebe-824a-129e0d348421:
ID       EXT RESOLUTION FPS │   TBR PROTO │ VCODEC        VBR ACODEC    ABR
───────────────────────────────────────────────────────────────────────────
hls-489  mp4 384x216     29 │  489k m3u8  │ avc1.4d401f  489k mp4a.40.2  0k
hls-847  mp4 512x288     29 │  847k m3u8  │ avc1.4d401f  847k mp4a.40.2  0k
hls-1251 mp4 640x360     29 │ 1251k m3u8  │ avc1.4d401f 1251k mp4a.40.2  0k
hls-1864 mp4 896x504     29 │ 1864k m3u8  │ avc1.4d401f 1864k mp4a.40.2  0k
hls-2604 mp4 960x540     59 │ 2604k m3u8  │ avc1.4d4020 2604k mp4a.40.2  0k
hls-3621 mp4 1280x720    59 │ 3621k m3u8  │ avc1.640028 3621k mp4a.40.2  0k
hls-5763 mp4 1280x720    59 │ 5763k m3u8  │ avc1.640028 5763k mp4a.40.2  0k
```

Additional test
```
{
        'url': 'https://www.espn.com/espnplus/player/_/id/317f5fd1-c78a-4ebe-824a-129e0d348421',
        'info_dict': {
            'id': '317f5fd1-c78a-4ebe-824a-129e0d348421',
            'ext': 'mp4',
            'title': 'The Wheel - Episode 10',
            'thumbnail': 'https://s.secure.espncdn.com/stitcher/artwork/collections/media/317f5fd1-c78a-4ebe-824a-129e0d348421/16x9.jpg?timestamp=202205031523&showBadge=true&cb=12&package=ESPN_PLUS',
        },
        'params': {
            'skip_download': True,
        },
    }
```
